libyul/optimiser/LoadResolver.cpp

```diff
@@ -76,3 +99,40 @@ void LoadResolver::tryResolve(
 			if (inScope(*value))
 				_e = Identifier{locationOf(_e), *value};
 }
+
+void LoadResolver::tryEvaluateKeccak(
+	Expression& _e,
+	std::vector<Expression> const& _arguments
+)
+{
+	auto valueOfIdentifier = [&](YulString const& _name) -> optional<u256>
+	{
+		if (m_value.count(_name))
+			if (Literal const* literal = get_if<Literal>(m_value.at(_name).value))
+				return valueOfLiteral(*literal);
+		return nullopt;
+	};
+
+	yulAssert(_arguments.size() == 2, "");
+	Identifier const* memoryKey =  std::get_if<Identifier>(&_arguments.at(0));
+	Identifier const* offset = std::get_if<Identifier>(&_arguments.at(1));
+
+	if (!memoryKey || !offset) return;
+
+	if (
+		auto memoryValue = util::valueOrNullptr(m_memory, memoryKey->name);
+		memoryValue &&
+		inScope(*memoryValue)
+	)
+	{
+		auto memoryContent = valueOfIdentifier(*memoryValue);
+		auto memoryOffset = valueOfIdentifier(offset->name);
+		if (memoryContent && memoryOffset == 32)
+			_e = Literal{
+				locationOf(_e),
+				LiteralKind::Number,
+				YulString{u256(keccak256(toBigEndian(*memoryContent))).str()},
```

Comment:
Is there any particular reason why you don't just truncate `toBigEndian(*memoryContent)` to a size of `memoryOffset` for `memoryOffset < 32`?

Reply:
Good idea. I'll implement it!